File: server/iih_pub/iih.ci_pub/src/main/java/iih/ci/mrqc/i/IMrqcMaintainService.java
package iih.ci.mrqc.i;

import xap.mw.core.data.BizException;
import xap.mw.coreitf.d.FBoolean;
import iih.ci.mrqc.qaflt.d.QaFltDTO;
import iih.ci.mrqc.qared.d.QaRecordDO;
import iih.ci.mrqc.revisionnotice.d.RevisionNoticeDO;

public interface IMrqcMaintainService {
	/**
	 * 保存整改通知
	 * @param revnotice
	 * @throws BizException
	 */
	public FBoolean SaveRevNotice(RevisionNoticeDO revnotice,QaFltDTO[] qaflt,QaRecordDO qarecord,String id_amr) throws BizException;

	/**
	 * 更新缺陷状态
	 * @param revnotice
	 * @throws BizException
	 */
	public FBoolean updateStatus(QaFltDTO[] qaflt) throws BizException;
}

File: client/iih_stub/iih.ci_stub/iih.ci.mrqc/i/IMrqcMaintainService.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using iih.ci.mrqc.qaflt.d;
using iih.ci.mrqc.qared.d;
using iih.ci.mrqc.revisionnotice.d;
using xap.mw.coreitf.d;
using iih.ci.mrqc.d;

namespace iih.ci.mrqc.i
{
    public interface IMrqcMaintainService
    {
        FBoolean saveRevNotice(RevisionNoticeDO revnotice, QaFltDTO[] qaflt, QaRecordDO qarecord,string id_amr);
        
        FBoolean updateStatus(QaFltDTO[] qaflt);

        Cidiagdto[] getCidiagdto(String id_ent);
    }
}
